### ISSUE TITLE
Proposal: Add Concentration for casters

### ISSUE THREAD

**Issue by MikeyLikes** [torilmud/issues#681]
I know we don't need to faithfully follow dnd rules, but this is potentially a way to improve casters and make them more viable.

Here's the idea:

1. Every caster has a set of spells that use concentration.
2. A caster can only maintain one concentrated spell at a time.
3. Casters can cast non-concentration spells while they are concentrating on a spell.
4. A new skill, "Concentration" is added for all casters that is based on Constitution.
5. Whenever a concentrating caster takes damage, there is a chance that they lose concentration. This would be a check against that concentration skill.
6. Getting bashed / stunned / etc would break concentration.

**Potential examples of concentration spells:**

Invoker: 

Inferno becomes concentration and does aoe damage every round while up. Invokers could additionally cast incendiary cloud which is not a concentrated spell.

2nd concentration skill could be a ray spell that ticks high damage single target. Invokers could layer on force missiles.

Enchanter: 

Concentration aura spells that affect the entire group giving them enlarge or reduce or blur. Scales and stone stay as single target.

Illusionist:

Concentration single target "counterspell" which significantly lowers mob concentration or even counters / reduces damage from casters.

Elementalist:

Stinking cloud concentration spell that has a chance every turn to weaken or even stun the room.
Summon becomes a concentrated spell that summons elementals for combat only which would allow us to make them significantly more powerful offensively.

Necro / Lich

Rot becomes concentration with per turn tick. 
Pets become are automatically raised during combat if soul bound and fall apart quickly after combat.

Druid

Creeping Doom is concentration
Can heal as power / action

Shaman

Group Heal becomes concentration so they can focus on hex / silence etc in combat

Cleric

Single target pulsing concentration heal. Layers on more healing as powers

Summary:

Concentration could be a great way to improve viability of casters in general and make their gameplay more interactive. It would also give every caster a reason to "be in the room" for combat and eliminate the flee / mem mechanics that everyone dislikes.

Over time this could be implemented for mobs and players.

**Comment by Inames** [comment]
I like where your head is at, but a lot of the ideas break functionality. like having to be in a room to maintain concentration spells on a group, or the possibility of key defensive spells falling because a mob hitalls and hits the chanter causing blur to drop, tanks to die, and group to spank. I think we need the kinds of spell upgrades and variation you are suggesting, but concentration as a mechanic in this game is absolutely going to fail because you can't play strategically to avoid damage or avoid being bashed etc to maintain your spells when mobs can literally attack anyone at any time for no logical reason.

**Comment by Inames** [comment]
Something that might be an interesting alternative, is creating powers that drain mana, which exists on all characters regardless of class, as something that can be toggled on/off and drain per "round." This could open up recovery song to being something useful again if its made to be better psp renewal for group. Also scaling that mana with the primary stat of the class, or making power ACTUALLY do something, would be good. Because as is, most races dont get more than like 30 unless you are snake, bard, or psi.

**Comment by MikeyLikes** [comment]
yeah, I'm not really advocating for this exact idea to be implemented, but I think most of the objections you mentioned could be tweaked. . .movement might not need to break conc.

I do think it's important that there is a chance to have concentration broken on a hitall or incendiary cloud. Restarting it wouldn't be too involved and would give casters something to do in combat beyond watching ******* -> **** -> *.

My driving force here is making casters more useful at the high end and more interesting to play.

Thanks for the feedback and additional ideas.

**Comment by shevarash** [comment]
Interesting ideas. I'll leave this open until the caster classes are overhauled and incorporate any ideas as needed. Thank you for the detailed and well written proposal.
